gossipd/routing.c

```diff
@@ -832,19 +810,23 @@ static struct pending_cannouncement *
 find_pending_cannouncement(struct routing_state *rstate,
 			   const struct short_channel_id *scid)
 {
-	struct pending_cannouncement *i;
+	struct pending_cannouncement_map_iter i;
+	struct pending_cannouncement *pann;
 
-	list_for_each(&rstate->pending_cannouncement, i, list) {
-		if (short_channel_id_eq(scid, &i->short_channel_id))
-			return i;
+	for (pann = pending_cannouncement_map_first(&rstate->pending_cannouncements, &i);
+	     pann;
+	     pann = pending_cannouncement_map_next(&rstate->pending_cannouncements, &i)) {
+		if (short_channel_id_eq(scid, &pann->short_channel_id))
+			return pann;
```

Comment:
Oops, this iterates through the hash table!  You want:

```
pann = pending_cannouncement_map_get(&rstate->pending_cannouncements, scid);
```

Reply:
I'm so Sorry...I corrected it!